src/main/java/com/conveyal/taui/controllers/BundleController.java

```diff
@@ -121,38 +128,47 @@ public static Bundle create (Request req, Response res) {
         // process async
         ExecutorServices.heavy.execute(() -> {
             try {
-                TDoubleList lats = new TDoubleArrayList();
-                TDoubleList lons = new TDoubleArrayList();
                 Set<String> seenFeedIds = new HashSet<>();
 
+                Envelope bundleBounds = new Envelope();
+                bundle.serviceStart = LocalDate.MAX;
+                bundle.serviceEnd = LocalDate.MIN;
                 bundle.feeds = new ArrayList<>();
                 bundle.totalFeeds = localFiles.size();
 
-
                 for (File file : localFiles) {
                     FeedSource fs = ApiMain.registerFeedSource(feed -> String.format("%s_%s", feed.feedId, bundle._id), file);
                     if (seenFeedIds.contains(fs.feed.feedId)) {
                         throw new Exception("Duplicate Feed ID found when uploading bundle");
                     }
 
-                    bundle.feeds.add(new Bundle.FeedSummary(fs.feed, bundle._id));
-                    fs.feed.stops.values().forEach(s -> {
-                        lats.add(s.stop_lat);
-                        lons.add(s.stop_lon);
-                    });
+                    Bundle.FeedSummary feedSummary  = new Bundle.FeedSummary(fs.feed, bundle._id);
+                    bundle.feeds.add(feedSummary);
+
+                    for (Stop s : fs.feed.stops.values()) {
+                        bundleBounds.expandToInclude(s.stop_lon, s.stop_lat);
+                    }
+
+                    if (bundle.serviceStart.isAfter(feedSummary.serviceStart)) {
+                        bundle.serviceStart = feedSummary.serviceStart;
+                    }
+
+                    if (bundle.serviceEnd.isBefore(feedSummary.serviceEnd)) {
+                        bundle.serviceEnd = feedSummary.serviceEnd;
+                    }
 
                     bundle.feedsComplete += 1;
                     Persistence.bundles.put(bundle);
                 }
 
-                // find the median stop location
-                // use a median because it is robust to outliers
-                lats.sort();
-                lons.sort();
-                // not a true median as we don't handle the case when there is an even number of stops
-                // and we are supposed to average the two middle value, but close enough
-                bundle.centerLat = lats.get(lats.size() / 2);
-                bundle.centerLon = lons.get(lons.size() / 2);
+                // TODO Handle crossing the antimeridian
+                bundle.north = bundleBounds.getMaxY();
+                bundle.south = bundleBounds.getMinY();
+                bundle.east = bundleBounds.getMaxX();
+                bundle.west = bundleBounds.getMinX();
+
+                bundle.centerLat = (bundle.north + bundle.south) / 2;
```

Comment:
The previous approach was robust to outliers -- even if there was a stray stop at null island, the center coordinate of the bundle would be in a sensible location.  Maybe we're okay doing away with that, because origin locations are set from region (not bundle) bounds, and we're just going to show the bounds of the bundle for debugging/troubleshooting purposes (with https://github.com/conveyal/analysis-backend/issues/148)

Reply:
Turns out we don't use this anymore in the code. Most likely because it can be easily calculated on the fly. I'm going to remove the code creating and storing it.

Reply:
I was going to make a similar comment to Anson, but if it's not used then I'd agree we should remove it. However, the original change was basically unrelated to the stated purpose of the PR. Though ideally we'd make each change in a separate PR, I'm not actually against "rider" commits just for efficiency's sake. But we should aim to always mention and explain them in the PR description, so they don't sneak in without review by mistake.